sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -547,6 +548,16 @@ object CollapseRepartition extends Rule[LogicalPlan] {
 }
 
 /**
+ * Combines adjacent [[RepartitionByExpression]] operators.
+ */
+object CollapseRepartitionBy extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan transformUp {
+    case RepartitionByExpression(exprs, RepartitionByExpression(_, child, _), numPartitions) =>
```

Comment:
Then we can consolidate `CollapseRepartition` and `CollapseRepartitionBy` into one rule